mindmeld/augmentation.py

```diff
@@ -241,6 +262,167 @@ def __init__(self, batch_size, language, paths, path_suffix, resource_loader):
             "max_length": 60,
         }
 
+    def _download_model(self):
+        logger.info("Downloading paraphrase model from %s", PARAPHRASER_RETAIN_ENTITIES_URL)
+
+        # Make the folder that will contain the model folder
+        if not os.path.exists(EMBEDDINGS_FOLDER_PATH):
+            os.makedirs(EMBEDDINGS_FOLDER_PATH)
+
+        with TqdmUpTo(unit="B", unit_scale=True, miniters=1, desc='') as t:
+            try:
+                urlretrieve(PARAPHRASER_RETAIN_ENTITIES_URL,
+                            PARAPHRASER_FILE_PATH,
+                            reporthook=t.update_to)
+            except ConnectionError as e:
+                logger.error("Model download failed with error: %s", e)
+                return
+        try:
+            with zipfile.ZipFile(PARAPHRASER_FILE_PATH, 'r') as zip_ref:
+                zip_ref.extractall(EMBEDDINGS_FOLDER_PATH)
+        except zipfile.BadZipfile:
+            logger.error("Unable to extract zip file. Try downloading the model again.")
+        return
+
+    def _prepare_inputs(self, queries):
```

Comment:
We can add this as an abstract function in the abstract augmentor class? Since it looks like it's being inherited by both the English and multi-lingual augmentors.

Reply:
Had some thoughts about this - We can move the method currently implemented in the Multi-Lingual Paraphraser (to return unannotated queries) as the default and let the inheritors override it if they want to. However, I'm not sure if that is the correct default (since in the case of the English paraphraser, we now do require those annotations). On the flip side, making it an abstract method will potential force replication of the method as is in the Multi-Lingual Paraphraser to other future inheritors (maybe language specific models(?)). 

What are your thoughts on what makes a good default? @vijay120 @KKsharma99 @ritvikshrivastava

Reply:
If we think future language specific models will be based on the similar MT strategy, then we should make the multi-lingual method as the abstract. 

Otherwise, the current implementation with an empty abstract is fine. We can re-visit this if and when another model is added in this case.

Reply:
If we do not expect inheritors to use this method as it is, we should avoid making it the default in the base class. A middle ground here is to see if we can refactor this function in a way that the logic we expect to be used by the inheritors is made into a separate function. The function with the common logic can be placed in the abstract class and any custom alterations on top can be done via an overridden method.

What aspects of this method would be specific to EN, is it the syntax of <unannotated text> <EOS> <entity values>?